build-system/cpp-cmake-conan/src/build.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import subprocess
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import List
+
+from shared_utils import get_valid_arch
+from velocitas_lib import get_workspace_dir
+
+CMAKE_EXECUTABLE = "cmake"
+CONAN_EXECUTABLE = "conan"
+
+
+def safe_get_workspace_dir() -> str:
```

Comment:
would be a candidate to move to the velocitas-lib?

Reply:
This is just in here b/c the basic scripts (`build.sh`, `install_dependencies.sh`) now proxy through to python and are not called through CLI. velocitas lib is supposed to be used with CLI only, hence no need to port this function over.